servicetalk-http-api/src/main/java/io/servicetalk/http/api/AbstractZipContentCodec.java

```diff
@@ -0,0 +1,558 @@
+                switch (state) {
+                    case FOOTER_START:
+                        if (readGZIPFooter(in)) {
+                            finished = true;
+                        }
+                        return null;
+                    default:
+                        if (state != State.HEADER_END && !readGZIPHeader(in)) {
+                            return null;
+                        }
+                }
+                // Some bytes may have been consumed, and so we must re-set the number of readable bytes.
+                readableBytes = in.readableBytes();
+            }
+
+            if (in.hasArray()) {
+                inflater.setInput(in.array(), in.arrayOffset() + in.readerIndex(), readableBytes);
+            } else {
+                byte[] array = new byte[readableBytes];
+                in.getBytes(in.readerIndex(), array);
+                inflater.setInput(array);
+            }
+
+            try {
+                boolean readFooter = false;
+                while (!inflater.needsInput()) {
+                    byte[] outArray = decompressed.array();
+                    int writerIndex = decompressed.writerIndex();
+                    int outIndex = decompressed.arrayOffset() + writerIndex;
+                    int outputLength = inflater.inflate(outArray, outIndex, decompressed.writableBytes());
+                    if (outputLength > 0) {
+                        decompressed.writerIndex(writerIndex + outputLength);
+                        if (crc != null) {
+                            crc.update(outArray, outIndex, outputLength);
+                        }
+                    } else {
+                        if (inflater.needsDictionary()) {
+                            throw new IOException(
+                                    "decompression failure, unable to set dictionary as non was specified");
+                        }
+                    }
+
+                    if (inflater.finished()) {
+                        if (crc == null) {
+                            finished = true; // Do not decode anymore.
+                        } else {
+                            readFooter = true;
+                        }
+                        break;
+                    } else {
+                        decompressed.ensureWritable(chunkSize);
+                    }
+                }
+
+                in.skipBytes(readableBytes - inflater.getRemaining());
+
+                if (readFooter) {
+                    state = State.FOOTER_START;
+                    if (readGZIPFooter(in)) {
+                        finished = true;
+                        inflater.end();
+                    }
+                }
+            } catch (DataFormatException e) {
+                throw new IOException("decompression failure", e);
+            }
+
+            if (decompressed.readableBytes() > 0) {
+                return decompressed.readSlice(decompressed.readableBytes());
```

Comment:
I suspect there is a similar issue with decompressed destination buffer.

Reply:
Same